framework/include/reporters/ConstantReporter.h

```diff
@@ -21,8 +21,50 @@ class ConstantReporter : public GeneralReporter
   virtual void execute() override {}
 
 protected:
+  /// Helper for declaring constant reporter values
+  template <typename T>
+  std::vector<T *> declareConstantReporterValues(const std::string & names_param,
+                                                 const std::string & values_param);
+
+  /// Integer reporter data
   std::vector<int *> _int;
+  /// Real reporter data
   std::vector<Real *> _real;
-  std::vector<std::vector<Real> *> _vector;
+  /// String reporter data
   std::vector<std::string *> _string;
+  /// Vector of integers reporter data
+  std::vector<std::vector<int> *> _int_vec;
+  /// Vector of reals reporter data
+  std::vector<std::vector<Real> *> _real_vec;
+  /// Vector of strings reporter data
+  std::vector<std::vector<std::string> *> _string_vec;
 };
+
+template <typename T>
+std::vector<T *>
+ConstantReporter::declareConstantReporterValues(const std::string & names_param,
+                                                const std::string & values_param)
```

Comment:
Do you think it is worth having this just take a prefix: "integer" and then build the names_param and values_param from it?